tests/test_adapted_attributes.py

```diff
@@ -0,0 +1,153 @@
+import os
+import pytest
+import tempfile
+import datajoint as dj
+from datajoint.errors import ADAPTED_TYPE_SWITCH, FILEPATH_FEATURE_SWITCH
+import networkx as nx
+from itertools import zip_longest
+from . import schema_adapted
+from .schema_adapted import Connectivity, Layout
+from . import PREFIX, S3_CONN_INFO
+
+
+@pytest.fixture
+def adapted_graph_instance():
+    yield schema_adapted.GraphAdapter()
+
+
+@pytest.fixture
+def enable_adapted_types(monkeypatch):
+    monkeypatch.setenv(ADAPTED_TYPE_SWITCH, 'TRUE')
+    yield
+    monkeypatch.delenv(ADAPTED_TYPE_SWITCH, raising=True)
+
+
+@pytest.fixture
+def enable_filepath_feature(monkeypatch):
+    monkeypatch.setenv(FILEPATH_FEATURE_SWITCH, 'TRUE')
+    yield
+    monkeypatch.delenv(FILEPATH_FEATURE_SWITCH, raising=True)
+
+
+@pytest.fixture
+def schema_name_custom_datatype():
+    schema_name = PREFIX + "_test_custom_datatype"
+    return schema_name
```

Comment:
Similar to `schema_adapted.GraphAdapter()`, we could just use a global variable here.

Reply:
We could make it a global var too. I put it in a fixture to avoid duplicating logic, because it's used by multiple other fixtures.

Reply:
I think it's less confusing as a global constant: `SCHEMA_NAME = PREFIX + "_test_custom_datatype"`